xonsh/environ.py

```diff
@@ -191,6 +194,7 @@ def xonshconfig(env):
     'PUSHD_SILENT': False,
     'RAISE_SUBPROC_ERROR': False,
     'RIGHT_PROMPT': '',
+    'DYNAMIC_CWD_WIDTH': '20',
```

Comment:
> What should be the string representation of infinite ? +INF 

That works.  There are a number of things that generate Python's internal representation for infinity when passed as an argument to `float`.  In fact, it looks like an capitalization of INF works, with or without the "+" sign in front (so `"inf"`, `"Inf"`, `"inF"`, `"INf"`, `"INF"` all work).

Because Python's repr for infinity is lower-case `"inf"`, I might vote for using that.

```
>>> float('inf')
inf
```

> Maybe it would not be good to replace the default cwd because this one if more complex to compute maybe not everyone is willing to pay this price

Do you have any data on how long it takes to compute?  I imagine it would end up not being _that_ different/noticeable for most prompts.

In particular, if the default length is infinity, we could "short-circuit" all the computation and just return the prompt as-is.